File: ms-java/src/main/java/com/jakubeeee/playground/msjava/organization/internal/WorkerEntity.java
package com.jakubeeee.playground.msjava.organization.internal;

import com.jakubeeee.playground.msjava.organization.Position;
import org.jetbrains.annotations.NotNull;

import javax.persistence.*;

import static java.util.Objects.requireNonNull;

@Entity
@Table(name = "WORKERS")
@DiscriminatorValue("WORKER")
public class WorkerEntity extends EmployeeEntity {

    @ManyToOne
    @JoinColumn(name = "MANAGER_ID", nullable = false)
    private ManagerEntity manager;

    protected WorkerEntity() {
        // no-arg constructor required by the JPA
    }

    protected WorkerEntity(@NotNull String lastName, @NotNull Position position, long salary, @NotNull DepartmentEntity department, @NotNull ManagerEntity manager) {
        super(requireNonNull(lastName), requireNonNull(position), salary, requireNonNull(department));
        this.manager = requireNonNull(manager);
    }

    public ManagerEntity manager() {
        return manager;
    }

}

File: ms-kotlin/src/main/kotlin/com/jakubeeee/playground/mskotlin/organization/internal/WorkerEntity.kt
package com.jakubeeee.playground.mskotlin.organization.internal

import com.jakubeeee.playground.mskotlin.organization.Position
import javax.persistence.*

@Entity
@Table(name = "WORKERS")
@DiscriminatorValue("WORKER")
open class WorkerEntity : EmployeeEntity {

    @ManyToOne
    @JoinColumn(name = "MANAGER_ID", nullable = false)
    open var manager: ManagerEntity? = null
        protected set

    protected constructor()

    constructor(lastName: String, position: Position, salary: Long, department: DepartmentEntity, manager: ManagerEntity)
            : super(lastName, position, salary, department) {
        this.manager = manager
    }

}